### DISCUSSION TITLE
Update nib package

### DISCUSSION THREAD

**Discussion by dresende** [stylus/stylus#2670] [General]
I know there's a repo for that but it's unmaintained. @iChenLei is it possible to release a new version of nib with package.json fixed? [problem here](https://github.com/stylus/nib/issues/354)

If you don't want to have that responsibility or don't have permission, who can we ask for help? If you need volunteers, I'm ok with doing it, just maintaining package dependencies and security issues.

**Comment by iChenLei** [comment]
I don't have the permission to publish new version of nib, @dresende. I tried my best to solve this problem before, but no echo from @notslang

### Only @tj and @notslang can publish nib package
![2022-03-23 18 19 11](https://user-images.githubusercontent.com/14012511/159676948-25c46f7b-48c2-4d85-ba49-3491d0cb59a5.png)

**Reply by dresende** [reply]
Thank you for the quick reply. Will try to reach them.

**Reply by iChenLei** [reply]
Maybe I can request npm publish permission from npm official ? I don't know if it's possible.

**Reply by dresende** [reply]
Not sure. TJ is out of reach for me, tried DM Sean on twitter. Will see if he replies. I mentioned you didn't have permission, perhaps he can fix that.

**Reply by iChenLei** [reply]
Anyway, thanks for your effort for `stylus` ecosystem. ❤️

**Reply by dresende** [reply]
It's my only pre-processor and don't want to replace it at all cost :)

**Reply by notslang** [reply]
@iChenLei - you should be invited to the nib package now.

**Reply by MHuiG** [reply]
I recommend using GithubActions to publish the new version to NPM

**Comment by MHuiG** [comment]
Is it possible to publish a new package named ~nib2~ @stylus/nib?

**Reply by MHuiG** [reply]
I can't wait to use it, so I published a version myself on npm, you can published one yourself, or use this one I published.

https://github.com/CoPoKo/nib/blob/master/.github/workflows/npm-publish.yml

https://www.npmjs.com/package/@copoko/nib

**Reply by iChenLei** [reply]
> `@stylus` npm scope is registered, we can't publish `@stylus/nib`

Good idea ! But how about `const nib = require("stylus/nib")`, export `nib` from stylus package.

**Reply by MHuiG** [reply]
That looks good

**Reply by broofa** [reply]
@iChenLei Definitely reach out to NPM support.  It's been a while since I had to deal with them for something like this but, in the past, I've found them to be pretty reasonable.

The main issue for them is just insuring the module continues to be maintained responsibly.  To that end, I've found it helps to provide ...

* a record of your attempts to communicate with the module owner
* means to verify your credibility

On that latter point, definitely make sure they know you're a maintainer of this project on Github, and that you have publish permissions for `stylus` already, but not for `nib`.

**Reply by iChenLei** [reply]
@broofa Thanks for your suggestion, I will try it.
